hmpps-kotlin-spring-boot-test-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/test/kotlin/auth/JwtAuthorisationHelper.kt

```diff
@@ -0,0 +1,81 @@
+package uk.gov.justice.hmpps.test.kotlin.auth
+
+import io.jsonwebtoken.Jwts
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type.REACTIVE
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type.SERVLET
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Primary
+import org.springframework.http.HttpHeaders
+import org.springframework.security.oauth2.jwt.JwtDecoder
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder
+import org.springframework.security.oauth2.jwt.NimbusReactiveJwtDecoder
+import org.springframework.security.oauth2.jwt.ReactiveJwtDecoder
+import org.springframework.stereotype.Component
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.interfaces.RSAPublicKey
+import java.time.Duration
+import java.util.Date
+import java.util.UUID
+
+@Component
+class JwtAuthorisationHelper {
```

Comment:
most of the time this is called `JwtAuthHelper` or `JwtAuthenticationHelper`, but opted for authorisation here as the header it sets is `AUTHORIZATION`